### PR TITLE
Moved TagHelperBinder from pass to a phase

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#1118 

@rynowak @NTaylorMullen

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs [hunk 1]
```diff
@@ -8,24 +8,23 @@
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
-    internal class TagHelperBinderSyntaxTreePass : IRazorSyntaxTreePass
+    internal class DefaultRazorTagHelperBinderPhase : RazorEnginePhaseBase, IRazorTagHelperBinderPhase
     {
         private static HashSet<char> InvalidNonWhitespaceNameCharacters = new HashSet<char>(new[]
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs]: Is this used? (It shouldn't be)

File: test/Microsoft.AspNetCore.Razor.Language.Test/DefaultRazorTagHelperBinderPhaseTest.cs [hunk 2]
```diff
@@ -883,11 +908,11 @@ public static TheoryData ProcessTagHelperPrefixData
         {
             // Arrange
             var errorSink = new ErrorSink();
-            var pass = new TagHelperBinderSyntaxTreePass();
+            var phase = new DefaultRazorTagHelperBinderPhase();
             var document = RazorCodeDocument.Create(new DefaultRazorSourceDocument("Test content", encoding: Encoding.UTF8, fileName: "TestFile"));
 
             // Act
-            var prefix = pass.ProcessTagHelperPrefix(((IEnumerable<TagHelperDirectiveDescriptor>)directiveDescriptors).ToList(), document, errorSink);
+            var prefix = phase.ProcessTagHelperPrefix(((IEnumerable<TagHelperDirectiveDescriptor>)directiveDescriptors).ToList(), document, errorSink);
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Language.Test/DefaultRazorTagHelperBinderPhaseTest.cs]: You can put `IEnumerable<TagHelperDirectiveDescriptor>` in the parameter list and get rid of this cast. This was an old workaround for when THD was internal.

File: src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs (hunk 1 above)
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs]: Yes it is. [Here](https://github.com/aspnet/Razor/blob/ajbaaska/tgbinder-phase/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs#L246)

File: src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs [hunk 3]
```diff
@@ -8,24 +8,23 @@
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
-    internal class TagHelperBinderSyntaxTreePass : IRazorSyntaxTreePass
+    internal class DefaultRazorTagHelperBinderPhase : RazorEnginePhaseBase, IRazorTagHelperBinderPhase
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs]: From a naming perspective I don't see the value in having `Razor` in this name. Do our other phases follow the inclusion of Razor?

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs]: Nm, I see they do. This is good as is 👍

**Pull request merged**